lib/bolt/task.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'bolt/error'
+
+module Bolt
+  class Task
```

Comment:
I used a struct at https://github.com/puppetlabs/bolt/blob/master/lib/bolt/applicator.rb#L7 for a similar purpose (though skipped validation because I controlled the task). It'd be nice to unify these eventually.

Reply:
I think for now it makes more sense to use a struct. I based this off of `Puppet::Module::Task` but since we're not loading the task from a module I don't think we'll need any of that functionality.